### PR TITLE
Fix compiler warnings treated as errors in GistEncodeDecode.

### PR DESCRIPTION
**Pull request by mrry**
**Description**: Fix compiler warning in GistEncodeDecode.

**Motivation and Context**
- Method argument `compression_type` was shadowing member variable `compression_type`, which is a warning-treated-as-error in Windows builds.

### CODE REVIEW

**Review by edgchen1 — DISMISSED** [review]

File: orttraining/orttraining/core/optimizer/gist_encode_decode.h [hunk 1]
```diff
@@ -34,14 +33,16 @@ class GistEncodeDecode : public RewriteRule {
...
+  GistEncodeDecode(int op_type, std::string compr_type) noexcept : RewriteRule("GistEncodeDecode"), operator_type(op_type), compression_type_(std::move(compr_type)) {}
 
  private:
   int GenerateDecodePriority() const { return priority_generator_--; };
   std::vector<std::string> TargetOpTypes() const noexcept override;
   bool SatisfyCondition(const Graph& graph, const Node& node, const logging::Logger& logger) const override;
   Status Apply(Graph& graph, Node& node, RewriteRuleEffect& rule_effect, const logging::Logger& logger) const override;
   bool AddEncodeDecode(Graph& graph, Node& curr_node, std::string compression_type, const logging::Logger& logger) const;
+
+  std::string compression_type_;
```
Comment (edgchen1) [inline orttraining/orttraining/core/optimizer/gist_encode_decode.h:45]: while we're updating this, can compression_type_ be const?

Comment (mrry) [reply orttraining/orttraining/core/optimizer/gist_encode_decode.h]: Good idea! Done.

**Comment by edgchen1** [comment]
Windows GPU build error, probably introduced by the same PR:
```
orttraining\orttraining\training_ops\cuda\gist\gist_impl.cu(355,0): Error : calling a __host__ function("__floorf") from a __global__ function("onnxruntime::cuda::_GistPackMsfp15DecoderKernel<float> ") is not allowed
```
maybe we can call `floorf` here instead? or static_cast<uint8_t> even?
https://github.com/microsoft/onnxruntime/blob/a647da3e1acb5ecb750c19df6b1b85af4572237d/orttraining/orttraining/training_ops/cuda/gist/gist_impl.cu#L355

**Comment by mrry** [comment]
> Windows GPU build error, probably introduced by the same PR:
> 
> ```
> orttraining\orttraining\training_ops\cuda\gist\gist_impl.cu(355,0): Error : calling a __host__ function("__floorf") from a __global__ function("onnxruntime::cuda::_GistPackMsfp15DecoderKernel<float> ") is not allowed
> ```
> 
> maybe we can call `floorf` here instead? or static_cast<uint8_t> even?

Gah! Switching to `floorf()`.

**Review by edgchen1 — DISMISSED** [review]

**Review by skottmckay — APPROVED** [review]

**Review by edgchen1 — APPROVED** [review]

**Comment by mrry** [comment]
Thanks for the approvals! I'm going to go ahead and merge this - even though some builds are still red - because it looks like Gist is now compiling successfully and there's a separate issue in orttraining\orttraining\training_ops\cuda\loss\softmaxcrossentropy_impl.cu on the Debug build :(

https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=379294&view=logs&j=589bd21c-4dcb-5f8b-bb33-cc0cd6a6211e&t=2945763a-df57-58d8-5a72-90a03892b775&l=19633

**Pull request merged**